clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsTest.java

```diff
@@ -1004,10 +998,6 @@ public void testWithRecords(Args args) {
         }
```

Comment:
Is ```AtLeastV2OrNotZstd``` more suitable to ```testWithRecords```?

Reply:
@chia7712 In my opinion , It checks the IllegalArgumentException( unsupport method ).
So I think it is necessary

Reply:
It is another test style - we can pass all parameters to all test cases and then explicitly call assertThrow to handle the unsupported method(s). This style can bring some benefits: 

1. Single parameter suite
2. explicitly show the “unsupported” method

WDYT?

Reply:
@chia7712  It seems like before . 
But I think that checking all `new  MemoryRecordsBuilder ` will throw by some args is not easy.

what do you think

Reply:
@g1geordie I file a patch for aforementioned idea. Please take a look at https://github.com/chia7712/kafka/pull/1/files

it uses explicit assert (exception or expected value) for all parameters instead of just ignoring them

Reply:
It sound like you change `assume (condition)` to  `if (condition) ... else ...`  in all method